TestVectorsAwsCryptographicMaterialProviders/runtimes/python/fuzz_generator.py

```diff
@@ -0,0 +1,338 @@
+    if test_type == "negative-encrypt-keyring":
+        test_vector["errorDescription"] = "Expected encryption failure"
+    elif test_type == "negative-decrypt-keyring":
+        test_vector["decryptErrorDescription"] = "Expected decryption failure"
+
+
+# Assembling the test vector
+
+@composite
+def fuzz_test_vector(draw):
+    """Generate a complete fuzzed test vector."""
+    # Generate basic components
+    encryption_context = draw(fuzz_encryption_context())
+    algorithm_suite = draw(st.sampled_from(ALGORITHM_SUITES))
+    test_type = draw(st.sampled_from(TEST_TYPES))
+    keyring_type = draw(st.sampled_from(KEYRING_TYPES))
+
+    if keyring_type in ["kms", "aws-kms-mrk-aware", "caching-cmm"]:
+        kms_key = draw(st.sampled_from(KMS_KEYS))
+    else:
+        kms_key = None  # Raw keyrings don't need this
+    
+    # Generate required keys based on test type
+    required_keys = generate_required_keys(draw, test_type, encryption_context)
+    
+    # Create key descriptions
+    key_description = create_key_description(draw, keyring_type, test_type, kms_key, required_keys)
+    
+    # Generate reproduced context
+    reproduced_context = generate_reproduced_context(draw, encryption_context)
+    
+    # Create test vector
+    test_vector = {
+        "type": test_type,
+        "description": get_description_template(test_type, keyring_type),
+        "algorithmSuiteId": algorithm_suite,
+        "encryptKeyDescription": key_description,
+        "decryptKeyDescription": key_description,
+        "reproducedEncryptionContext": reproduced_context,
+        "requiredEncryptionContextKeys": required_keys,
+        "encryptionContext": encryption_context
+    }
+    
+    # Add error descriptions for negative tests
+    add_error_descriptions(test_vector, test_type, keyring_type)
+    
+    return test_vector
+
+
+def extract_new_keys(test_vectors: Dict[str, Any]) -> Dict[str, Any]:
+    """Extract new keys from raw keyring test vectors.
+
+    - Scans all generated test vectors
+    - Finds raw keyring tests (type == "raw")
+    - Extracts the fuzzed key names they reference
+    - Creates corresponding key material entries for keys.json
+    - Returns a dict of new keys to add to keys.json
+    """
+    new_keys = {}
+    
+    for test_vector in test_vectors.values():
+        if test_vector.get("encryptKeyDescription", {}).get("type") == "raw":
+            key_name = test_vector["encryptKeyDescription"]["key"]
+            
+            # Determine base key type from key name
+            base_key_id = "aes-256"  # default
+            for key_type in RAW_KEY_TYPES:
+                if key_type in key_name:
+                    base_key_id = key_type
+                    break
+            
+            key_info = KEY_MATERIALS.get(base_key_id, KEY_MATERIALS["aes-256"])
+            new_keys[key_name] = {
+                "encrypt": True, "decrypt": True, "algorithm": "aes", "type": "symmetric",
```

Comment:
Put a TODO for algorithm, type, encoding, and anything that does not cover every cases.